### PR TITLE
Add Eslint to repo

### LINKED ISSUE #116: Add Eslint to repo
Since we are using prettier for the precommit hook I think we should add eslint aswell.

I was thinking of adding:

  * [eslint-config-airbnb](https://www.npmjs.com/package/eslint-config-airbnb)
  * [eslint-config-prettier](https://www.npmjs.com/package/eslint-config-prettier)
  * [eslint-plugin-compat](https://www.npmjs.com/package/eslint-plugin-compat)
  * [eslint-plugin-import](https://www.npmjs.com/package/eslint-plugin-import)
  * [eslint-plugin-prettier](https://www.npmjs.com/package/eslint-plugin-prettier)

How does that sound ?

### PR DESCRIPTION
**Pull request by SaraVieira**
Closes #116

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+  extends: ['prettier', 'airbnb-base'],
+  plugins: ['prettier', 'import'],
```
Comment (SaraVieira) [inline .eslintrc.js]: Added import for making sure we don't mistype file names

File: package.json [hunk 2]
```diff
@@ -51,12 +52,18 @@
   ],
   "lint-staged": {
     "*.js": [
-      "prettier --write --single-quote --no-semi",
+      "eslint --fix",
```
Comment (SaraVieira) [inline package.json]: Changed because now it does all the prettier fixes 😄

**Comment by karanjthakkar** [comment]
I was wondering if we should be keeping both `package-lock.json` and `yarn.lock` in the repo? What do you think w.r.t keeping them in sync?

File: .eslintrc.js (hunk 1 above)
Comment (ForsakenHarmony) [reply .eslintrc.js]: node is also an option ( the plugin )

**Comment by ForsakenHarmony** [comment]
@kakadiadarpan #114

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  rules: {
+    semi: 0,
```
Comment (ForsakenHarmony) [inline .eslintrc.js]: ok wait, if we don't want semis we could've used standard ¯\_(ツ)_/¯

**Comment by siddharthkp** [comment]
> I was wondering if we should be keeping both package-lock.json and yarn.lock in the repo? What do you think w.r.t keeping them in sync?

Nu! Extra step for contributors 😢

npm ftw!

**Comment by karanjthakkar** [comment]
> Nu! Extra step for contributors 😢

my feelings, exactly 😄

File: .eslintrc.js (hunk 1 above)
Comment (SaraVieira) [reply .eslintrc.js]: I'll do the switch , node makes more sense thanks Ashley

File: .eslintrc.js (hunk 3 above)
Comment (SaraVieira) [reply .eslintrc.js]: That's why I was thinking about standart :(

Changing it :)

**Comment by SaraVieira** [comment]
@ForsakenHarmony Fixed :)

File: .eslintrc.js (hunk 1 above)
Comment (ForsakenHarmony) [reply .eslintrc.js]: you could keep both, oh well

File: src/reporter.js [hunk 4]
```diff
@@ -6,12 +6,32 @@ const api = require('./api')
...
+  debug('global message', globalMessage)
+}
+
 const compare = (files, masterValues = {}) => {
   let fail = false
   let globalMessage
 
   files.map(file => (file.master = masterValues[file.path]))
-
   files.map(file => {
```
Comment (kuldeepkeshwar) [inline src/reporter.js]: shouldn't we combine the 2 loops here?
I know this is not related to Eslint

Comment (SaraVieira) [reply src/reporter.js]: WORD ! Fixed 🙂

**Comment by reznord** [comment]
guys, do we really need `package-lock.json`. Because of this, we are up to adding extra bloat to the project eg: `danger.js`

**Comment by reznord** [comment]
@SaraVieira the build failed on travis for the PR. `danger.js` bites really hard.

**Comment by SaraVieira** [comment]
On hold untill I get an answer from orta at dangerjs

**Comment by ForsakenHarmony** [comment]
imo we should've put the danger pr on hold

**Comment by ForsakenHarmony** [comment]
and not add babel for everything else now

**Comment by SaraVieira** [comment]
We are not using babel , we just need it in the deps for now , this will be removed, I don't even have a babelrc file

**Pull request merged**